src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -110,44 +121,126 @@ protected override void Dispose(bool disposing)
 
         #region Private methods
 
-        private void StartRequest(RequestCompletionSource state)
+        private async void StartRequest(RequestCompletionSource state)
```

Comment:
Bad combination of `async void`.
Seems there is no need in TaskCompletionSource at all and this could be

    private async Task<HttpResponseMessage> StartRequest(Request state)